### PR TITLE
[Taxon] Change taxon serialization group

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | no |
| License         | MIT |

The taxon ajax index endpoint returns way too much information about taxons.

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/Resources/config/routing/ajax/taxon.yml [hunk 1]
```diff
@@ -4,7 +4,24 @@ sylius_admin_ajax_taxon_index:
     defaults:
         _controller: sylius.controller.taxon:indexAction
         _sylius:
-            grid: sylius_admin_taxon
+            serialization_groups: [Ajax]
+            repository:
+                method: searchByName
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Resources/config/routing/ajax/taxon.yml]: We shoudl keep naming convention of repositories: `searchByName` -> `findByNamePart` or `findByPartOfName` ?

File: src/Sylius/Bundle/AdminBundle/Resources/config/routing/ajax/taxon.yml [hunk 2]
```diff
@@ -4,7 +4,24 @@ sylius_admin_ajax_taxon_index:
...
+                arguments:
+                    - $name
+                    - expr:service('sylius.context.locale').getLocaleCode()
+
+sylius_admin_ajax_taxon_in_criteria:
+    path: /in
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Resources/config/routing/ajax/taxon.yml]: This route should be default as it is behaviour of typical index

File: src/Sylius/Bundle/UiBundle/Resources/private/js/sylius-auto-complete.js [hunk 3]
```diff
@@ -11,82 +11,76 @@
...
-            $(this).each(function () {
-                var element = $(this);
-                var criteriaType = $(this).data('criteria-type');
-                var criteriaName = $(this).data('criteria-name');
-                var remoteUrl = $(this).data('url');
-                var choiceName = $(this).data('choice-name');
-                var choiceValue = $(this).data('choice-value');
-                var autocompleteValue = $(this).find('input.autocomplete').val();
+        autoComplete: function () {
+                $(this).each(function () {
```
Comment (pamil) [inline src/Sylius/Bundle/UiBundle/Resources/private/js/sylius-auto-complete.js]: I guess it should be less indented.

**Comment by Arminek** [comment]
It was merged in https://github.com/Sylius/Sylius/pull/7612

**Pull request closed without merging**